daemons/ipa-kdb/ipa_kdb_delegation.c

```diff
@@ -106,20 +106,26 @@ static krb5_error_code ipadb_match_acl(krb5_context kcontext,
     bool client_missing;
     bool client_found;
     bool target_found;
+    bool check_constraint_delegation = false;
+    size_t nrules = 0;
     int ret;
 
     ipactx = ipadb_get_context(kcontext);
     if (!ipactx) {
         return KRB5_KDB_DBNOTINITED;
     }
 
-    kerr = krb5_unparse_name(kcontext, client, &client_princ);
-    if (kerr != 0) {
-        goto done;
-    }
-    kerr = krb5_unparse_name(kcontext, target, &target_princ);
-    if (kerr != 0) {
-        goto done;
+    if ((client != NULL) && (target != NULL)) {
+        kerr = krb5_unparse_name(kcontext, client, &client_princ);
```

Comment:
`kerr` is not initialized but set unconditionally in the middle of the function. I think if would be safer to define it with an initial value of `ENOENT` and remove this assignation in the middle of the function.

Given the fact `kerr` is return systematically at the end of the function, we should know what is its default value by looking at its definition without having to go through the entire function.

Reply:
I withdraw this one. The `kerr` assignment in the middle of the function is actually necessary in order to reset the *failing by default* state after success exit code assignments from `krb5_unparse_name()` calls.